src/pymor/models/transforms.py

```diff
@@ -0,0 +1,171 @@
+# This file is part of the pyMOR project (http://www.pymor.org).
+# Copyright pyMOR developers and contributors. All rights reserved.
+# License: BSD 2-Clause License (http://opensource.org/licenses/BSD-2-Clause)
+
+import numpy as np
+from scipy.linalg import null_space
+
+from pymor.core.base import ImmutableObject
+
+
+class MoebiusTransformation(ImmutableObject):
+    r"""Maps the Riemann sphere onto itself.
+
+    A Moebius transformation
+
+    .. math::
+        M(s) = \frac{as+b}{cs+b}
+
+    is determined by the coefficients :math:`a,b,c,d\in\mathbb{C}`. The Moebius transformations form
+    a group under composition, therefore the `__matmul__` operator is defined to yield a
+    |MoebiusTransformation| if both factors are |MoebiusTransformation|s.
+
+    Parameters
+    ----------
+    coefficients
+        A tuple, list or |NumPy array| containing the four coefficients `a,b,c,d`.
+    normalize
+        If `True`, the coefficients are normalized, i.e. :math:`ad-bc=1`. Defaults to `False`.
+    name
+        Name of the transformation.
+    """
+
+    def __init__(self, coefficients, normalize=False, name=None):
+        assert len(coefficients) == 4
+        coefficients = np.array(coefficients)
+        assert coefficients[0]*coefficients[3] != coefficients[1]*coefficients[2]
+
+        if normalize:
+            coefficients = coefficients / np.sqrt(coefficients[0]*coefficients[3]-coefficients[1]*coefficients[2])
```

Comment:
```suggestion
            coefficients /= np.sqrt(coefficients[0]*coefficients[3]-coefficients[1]*coefficients[2])
```

Reply:
Hmm, I had to change this line because your suggestion gave me the `TypeError`


` TypeError: ufunc 'true_divide' output (typecode 'd') could not be coerced to provided output parameter (typecode 'l') according to the casting rule ''same_kind''`

When executing the tests.

Reply:
I like the idea. As far as I can tell, the normalization of coefficients is not too common anyways, so I would say its fine to have our own interpretation. I have put the absolute value inside the square root. This changes the result for complex values but it is still valid.